controllers/clusterresources/postgresqluser_controller.go

```diff
@@ -185,27 +262,11 @@ func (r *PostgreSQLUserReconciler) createUser(
 	newPassword string,
 	defaultUserSecretNamespacedName clusterresourcesv1beta1.NamespacedName,
 ) error {
-	defaultUserSecret := &k8sCore.Secret{}
-
-	namespacedName := types.NamespacedName{
-		Namespace: defaultUserSecretNamespacedName.Namespace,
-		Name:      defaultUserSecretNamespacedName.Name,
-	}
-	err := r.Get(ctx, namespacedName, defaultUserSecret)
+	defaultCreds, clusterName, err := r.getDefaultPostgreSQLUserCreds(ctx, defaultUserSecretNamespacedName)
 	if err != nil {
-		if k8sErrors.IsNotFound(err) {
-			return fmt.Errorf("cannot get default PostgreSQL user secret, user reference: %v, err: %w", defaultUserSecretNamespacedName, err)
-		}
-
 		return err
 	}
 
-	defaultUsername, defaultPassword, err := getUserCreds(defaultUserSecret)
-	if err != nil {
-		return fmt.Errorf("cannot get default PostgreSQL user credentials, user reference: %v, err: %w", defaultUserSecretNamespacedName, err)
-	}
-
-	clusterName := defaultUserSecret.Labels[models.ControlledByLabel]
```

Comment:
I think these are unnecessary changes. The old implementation was more straightforward and readable.
Now for example, why does the `r.getDefaultPostgreSQLUserCreds` return a cluster name?

Reply:
This code is duplicated in some places, so I decided to move it into a separate function. We fetch this creds by defaultUserSecretNamespacedName and I think we need to specify for what cluster they are for because for each cluster they are different